index.js

```diff
@@ -74,7 +74,9 @@ function vhost (hostname, handle) {
  */
 
 function hostnameof (req) {
-  var host = req.headers.host
+  var host = req.hostname || // express v4
+    req.host || // express v3
```

Comment:
`req.host`, despite the name, actually returns the hostname in Express.js 3. This has the same issue as noted above with the Express.js 4 `req.hostname`. It may also be important to note that in Express.js 5 `req.host` will actually return the host instead of the hostname like in 3/4. May need to determine best course of action on this too.